data_steward/gcloud/bq/__init__.py

```diff
@@ -184,3 +183,86 @@ def define_dataset(self, dataset_id: str, description: str,
         dataset.location = "US"
 
         return dataset
+
+    def copy_datasets(self, input_dataset: str, output_dataset: str):
+        """
+        Copies tables from source dataset to a destination datasets
+
+        :param input_dataset: name of the input dataset
+        :param output_dataset: name of the output dataset
+        :return:
+        """
+        # Copy input dataset tables to backup and staging datasets
+        tables = super(BigQueryClient, self).list_tables(input_dataset)
+        for table in tables:
+            staging_table = f'{output_dataset}.{table.table_id}'
+            self.copy_table(table, staging_table)
+
+    def list_tables(
+        self, dataset: bigquery.DatasetReference
+    ) -> typing.Iterator[bigquery.table.TableListItem]:
+        """
+        List all tables in a dataset
+
+        NOTE: Ensures all results are retrieved by first getting total
+        table count and setting max_results in list tables API call
+
```

Comment:
It think it would be good to add the behavior without _MAX_RESULTS_PADDING as a note, and why it is needed.

Reply:
So, keep the method as it is, and just remove lines 207 and 208?

Reply:
I'd say keep the method as is, and keep lines 207 and 208, but also add "Without setting max_results, the API has a bug causing it to randomly return 0 tables"